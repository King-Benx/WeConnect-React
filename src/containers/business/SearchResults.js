import React from 'react';
import { Button, Table} from 'react-bootstrap';
import axios from 'axios';
import { BASE_URL } from '../../custom/constants';
import CustomFunctions from '../../custom/CustomFunctions';
import DashboardNavigation from '../../navigation/DashboardNavigation';

class SearchResults extends React.Component{
    
    constructor(props){
        super(props);
        this.state = {
            all_businesses:[],
            q:props.match.params.search,
            filter_type:props.match.params.filter_type,
            filter_value:props.match.params.filter_value,
        }
        this.componentDidMount()
    }
    
    showReviews = (event) => {
        // handles redirect to the reviews of particular business
        this.props.history.push('/all_businesses/'+ event.target.id +'/reviews');
    }

    componentDidMount(){
        // populates the table with search results
        axios
        .get(BASE_URL+'api/v1/businesses/search?q='+this.state.q+'&filter_type='+ this.state.filter_type +'&filter_value='+ this.state.filter_value,
        {headers:{ 'x-access-token':CustomFunctions.getToken() }})
        .then(res => {
            if (res.status === 404){
                this.setState({ all_businesses:[] });
            }else if(res.status === 200){
                this.setState({ all_businesses:res.data.results.searched_businesses });
            }
        })
        .catch(err =>{
            CustomFunctions.createNotifications(err.status, err.response.message);
        });
    }
    render(){
        const table_data= this.state.all_businesses.map(((business)=>{
            return(
                <tr key={ business.id.toString() }>
                    <td>{ business.date_created }</td>
                    <td>{ business.name }</td>
                    <td>{ business.location }</td>
                    <td>{ business.category }</td>
                    <td>{ business.description }</td>
                    <td>
                        <Button id={business.id} onClick={ this.showReviews } bsStyle="info" block>
                            <i className="glyphicon glyphicon-info-sign" id={business.id} onClick={ this.showReviews }></i> View
                        </Button>
                    </td>
                </tr>
            );
        }));
        return (
            <div className="row"> 
                <DashboardNavigation/>
                <div className="col-sm-9 content-wrapper">
                    <div className="panel-heading">
                        <h3 className="panel-title">
                            <b>Search Results</b>
                        </h3>
                    </div>
                    <div className="panel panel-default">
                        <div className="panel-body">
                            <div className="table-responsive">
                                <Table striped bordered condensed hover>
                                    <thead>
                                        <tr>
                                            <th>Date of Creation</th>
                                            <th>Name</th>
                                            <th>Location</th>
                                            <th>Category</th>
                                            <th>Description</th>
                                            <th>View</th>
                                        </tr>
                                    </thead>
                                    <tbody>
                                        {table_data}
                                    </tbody>
                                </Table>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default SearchResults;