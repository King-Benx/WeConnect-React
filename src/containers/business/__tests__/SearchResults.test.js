import React from "react";
import { shallow } from "enzyme";
import SearchResults from "../SearchResults";
import mockAxios from "axios";

describe("These are tests for the search form ", () => {
  const props = {
    match: {
      params: {}
    }
  };
  const event = {
    target: {
      value: {}
    },
    preventDefault: () => {}
  };

  let mountedSearchResults;
  beforeEach(() => {
    mountedSearchResults = shallow(
      <SearchResults
        {...props}
        history={{
          push: () => {}
        }}
      />
    );
    mockAxios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: {
          results: {
            searched_businesses: [
              {
                category: "Information Technology",
                created_by: "admin",
                date_created: "Tue, 10 Jul 2018 12:27:46 GMT",
                description: "business 1 description",
                id: 1,
                last_modified: "Tue, 10 Jul 2018 12:27:46 GMT",
                location: "Kampala",
                name: "business 1",
                user_id: 1
              }
            ]
          }
        }
      })
    );
  });

  it("renders without crashing", () => {
    shallow(<SearchResults {...props} />);
  });

  it("handles show reviews", () => {
    mountedSearchResults.instance().showReviews(event);
  });

  it("has a dashboard", () => {
    const dashboards = mountedSearchResults.find("DashboardNavigation");
    expect(dashboards.length).toBe(1);
  });

  it("has a table", () => {
    const tables = mountedSearchResults.find("Table");
    expect(tables.length).toBe(1);
  });

  it("has a div", () => {
    const divs = mountedSearchResults.find("div");
    expect(divs.length).toBe(6);
  });

  it("has a table head", () => {
    const tableheads = mountedSearchResults.find("thead");
    expect(tableheads.length).toBe(1);
  });

  it("has a table body", () => {
    const tablebody = mountedSearchResults.find("tbody");
    expect(tablebody.length).toBe(1);
  });

  it("has an h3", () => {
    const h3s = mountedSearchResults.find("h3");
    expect(h3s.length).toBe(1);
  });

  it("has a b", () => {
    const bs = mountedSearchResults.find("b");
    expect(bs.length).toBe(1);
  });
});
