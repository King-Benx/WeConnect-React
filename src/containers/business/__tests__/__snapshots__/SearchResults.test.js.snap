// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the search form  renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 content-wrapper"
  >
    <div
      className="panel-heading"
    >
      <h3
        className="panel-title"
      >
        <b>
          Search Results
        </b>
      </h3>
    </div>
    <div
      className="panel panel-default"
    >
      <div
        className="panel-body"
      >
        <div
          className="table-responsive"
        >
          <table
            className="table table-striped table-bordered table-condensed table-hover"
          >
            <thead>
              <tr>
                <th>
                  Date of Creation
                </th>
                <th>
                  Name
                </th>
                <th>
                  Location
                </th>
                <th>
                  Category
                </th>
                <th>
                  Description
                </th>
                <th>
                  View
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
